### PR TITLE
Fix playback of HDR AV1 videos in native viewer

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/7973
* Part of https://github.com/rerun-io/rerun/issues/7594

Review with ignoring whitespace changes

#### Before
Screenshot 2024-11-02 at 12 02 13

#### After
Screenshot 2024-11-02 at 12 00 59

#### VLC (for reference)
Screenshot 2024-11-02 at 16 43 10

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7978?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7978?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7978)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/store/re_video/src/decode/av1.rs [hunk 1]
```diff
@@ -121,98 +122,115 @@ impl SyncDav1dDecoder {
...
+    let bits_per_component = picture
+        .bits_per_component()
+        .map_or(picture.bit_depth(), |bpc| bpc.0);
+
+    let bytes_per_component = if bits_per_component == 8 {
+        1
+    } else if 8 < bits_per_component && bits_per_component <= 16 {
+        // TODO(#7594): Support HDR video.
+        // We currnelty handle HDR videos by throwing away the lowest bits,
+        // and doing so rather slowly on CPU. It works, but the colors won't be perfectly correct.
```
Comment (Wumpf) [inline crates/store/re_video/src/decode/av1.rs:137]:
```suggestion
        // We currently handle HDR videos by throwing away the lowest bits,
        // and doing so rather slowly on CPU. It works, but the colors won't be perfectly correct.
```

**Review by Wumpf — APPROVED** [review]
nice! I trust there's still a separate warning about Bt.2020 interpretation and friends?
After all the problem with typical HDR content is twofold:
* bit depth - ~partially~ better addressed here
* color space conversion

**Comment by Wumpf** [comment]
oh interesting, the video in the issue is unplayable by Firefox. It has 10bit Bt.709 YUV 4:2:0
So indeed really the only thing we don't support here is the 10bits. Fascinating, didn't know that's a thing!

image

**Comment by emilk** [comment]
Bye @Wumpf we are missing a check that the number of bytes in the frame matches the given dimensions and pixel format. A debug-assert and/or warning in re_renderer would be very helpful

**Comment by Wumpf** [comment]
👍 good find, we should definitely add that

**Comment by emilk** [comment]
Actually, we do have a check here that triggers as it should:

https://github.com/rerun-io/rerun/blob/325fc41018c5eb75fbd18d9c84e9648ab88e4d70/crates/viewer/re_renderer/src/resource_managers/image_data_to_texture.rs#L186-L192

My bad!

**Comment by emilk** [comment]
Why does `rerun_js` fail?? O.o

**Pull request merged**